core/src/main/java/org/apache/calcite/sql/SqlDialect.java

```diff
@@ -754,6 +754,7 @@ public boolean supportsGroupByLiteral() {
 
   public boolean supportsAggregateFunction(SqlKind kind) {
     switch (kind) {
+    case LITERAL_AGG:
```

Comment:
this change does not seem related to the jdbc adapter
are there other aggregate functions that have to be considered for this list?
maybe this question is not related really to the bug you are solving

Reply:
This determines which aggregation functions can be pushed down, It is used in JdbcRules#canImplement. If this aggregation function can be pushed down, SqlImplementor#toSql will make `litetal_agg(literal)` to literal.

Reply:
True, but this change will have other effects besides enabling push-down.
But I guess this is fine. You are solving a more general problem than the issue states, but you have discovered through this particular issue.